### PR TITLE
remove early-exit from cleanup_nodes that broke dead node cleanup

### LINKED ISSUE #456: reimaging of dead nodes only happens if there are other nodes to get reimaged
Because of an early-exit in `cleanup_nodes`, specifically:
https://github.com/microsoft/onefuzz/blob/c0a4b0dba484bec3c4e97a52b4f69bd62061f3ad/src/api-service/__app__/onefuzzlib/pools.py#L892-L894

The `get_dead_nodes` call later on never gets called:
https://github.com/microsoft/onefuzz/blob/c0a4b0dba484bec3c4e97a52b4f69bd62061f3ad/src/api-service/__app__/onefuzzlib/pools.py#L907-L910

### PR DESCRIPTION
**Pull request by bmc-msft**
Fixes #456

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/pools.py [hunk 1]
```diff
@@ -921,10 +913,11 @@ def cleanup_nodes(self) -> bool:
 
             if to_reimage:
                 self.reimage_nodes(to_reimage)
+            return bool(to_reimage) or bool(to_delete)
```
Comment (chkeita) [inline src/api-service/__app__/onefuzzlib/pools.py:916]: isn't this over indented ?

File: src/api-service/__app__/onefuzzlib/pools.py [hunk 2]
```diff
@@ -889,10 +885,6 @@ def cleanup_nodes(self) -> bool:
 
         nodes_to_reset = [x for x in nodes if x.state in NodeState.ready_for_reset()]
```
Comment (chkeita) [inline src/api-service/__app__/onefuzzlib/pools.py:886]: i suggest rename `nodes` here to help the readability

File: src/api-service/__app__/onefuzzlib/pools.py (hunk 1 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/pools.py]: Yes.  Addressed.

File: src/api-service/__app__/onefuzzlib/pools.py (hunk 2 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/pools.py:886]: Not sure what you mean.  Can you provide more context?

**Review by chkeita — APPROVED** [review]

File: src/api-service/__app__/onefuzzlib/pools.py (hunk 2 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/pools.py]: Handled after offline discussion

**Pull request merged**